app/serializers/search/data_file_serializer.rb

```diff
@@ -0,0 +1,27 @@
+class Search::DataFileSerializer < ActiveModel::Serializer
+  attributes :id,
+             :name,
+             :is_deleted,
+             :created_at,
+             :updated_at,
+             :label,
+             :meta
+
+  has_many :tags, serializer: Search::TagSummarySerializer
+  has_one :project, serializer: Search::ProjectSummarySerializer
+  has_one :parent, serializer: Search::FolderSummarySerializer
+  has_one :creator, serializer: Search::UserSummarySerializer
+  
+  def is_deleted
+    object.is_deleted?
+  end
+
+  def meta
```

Comment:
Could the `meta` method possibly work as a has_many? Not a criticism, I'm honestly curious.